core/framework/cli/Command.php

```diff
@@ -268,11 +268,7 @@ static public function getOS() {
 
         public static function cli_echo($text, $color = 'white', $style = null)
         {
-            if (self::getOS() === 'OS_WIN' || self::getOS() === 'OS_UNKNOWN')
-            {
-                $return_text = $text;
-            }
-            else
+            if (getenv("TBG_NO_COLOR") != 1 && self::getOS() !== 'OS_WIN' && self::getOS() !== 'OS_UNKNOWN' && function_exists('posix_isatty') && posix_isatty(STDOUT))
```

Comment:
Can't we extract this into a separate function `useColors()` that returns early on each of the conditions?

Reply:
Hm... Most likely yes (should be cacheable as well), I'll get to it and update the PR.

Reply:
Ok, updated and rebased.